arrow-schema/src/schema.rs

```diff
@@ -45,6 +71,18 @@ impl SchemaBuilder {
         }
     }
 
+    /// Clears any fields currently in this builder.
+    pub fn clear_fields(mut self) -> Self {
```

Comment:
```suggestion
    pub fn clear(mut self) -> Self {
```

I can see the desire to distinguish this from metadata, but given the other fields aren't `push_field`, etc... Maybe this is more consistent?